docs/rules/strict.md

```diff
@@ -38,6 +38,8 @@ In the **CommonJS** module system, a hidden function wraps each module and limit
 
 In **ECMAScript** modules, which always have strict mode semantics, the directives are unnecessary.
 
+Since **ECMAScript 2016**, `"use strict"` directive in a function with non-simple parameter list is a syntax error. This rule disallows such `"use strict"` directives. See also the examples of [function option](#function)
```

Comment:
An alternative to consider: move the paragraph from the end of the introduction to the end of Rule Details and reword something like the following to be parallel with the other exception to report problems no matter which option:

This rule disallows strict mode directives, no matter which option is specified, in functions with non-simple parameter lists (for example, parameter lists with default parameter values) because that is a syntax error in **ECMAScript 2016** and later. See the examples of the [function](#function) option.